srcs/app/http/refactor/HttResponseFactory.cpp

```diff
@@ -0,0 +1,97 @@
+#include "HttpResponseFactory.hpp"
+
+HttpResponseFactory::HttpResponseFactory(
+		const std::string raw_request,
+		const ServerConfig &server_config
+	): server_config_(server_config) {
+	request_ = new HttpRequestDTO(raw_request);
+	request_method_ = request_->GetMethod();
+	concreteResponses_.insert(std::make_pair(
+		"GET",
+		&HttpResponseFactory::createHttpGetResponse_
+	));
+	concreteResponses_.insert(std::make_pair(
+		"POST",
+		&HttpResponseFactory::createHttpPostResponse_
+	));
+	concreteResponses_.insert(std::make_pair(
+		"DELETE",
+		&HttpResponseFactory::createHttpDeleteResponse_
+	));
+}
+
+HttpResponseFactory::~HttpResponseFactory() {
+}
+
+IResponse *HttpResponseFactory::response() {
+	if (request_ == NULL || request_->GetState() != RequestState::kComplete) {
+		return createHttpErrorResponse_(400);
+	}
+	if (!server_config_.common.return_url.empty()) {
+		return createHttpRedirectionResponse_();
+	}
+
+	request_location_ = new RequestLocation(server_config_, request_->GetPath());
+	if (!request_location_->common.return_url.empty()) {
+		return createHttpRedirectionResponse_();
+	} else if (HasAcceptedFormat_()) {
+		std::map<std::string, responseCreatorMethod>::const_iterator it;
+		it = concreteResponses_.find(request_method_);
+		if (it != concreteResponses_.end()) {
+			return CALL_MEMBER_FN(*this, it->second)();
+		} else {
+			return createHttpErrorResponse_(501);
```

Comment:
If I'm not mistaken, this will allocate memory on the heap, and it will be a bit difficult to check when we have to delete it.  The same thing applies to `createHttpRedirectionResponse_()` and `createHttpErrorResponse_()`.

For this case, maybe we could add a new method to the map, which will return a 501 Not Implemented response by default.

e.g.: `createHttpNotImplementedResponse_();`

Reply:
You are totally right about this. The thing this is a half implementation (you've probably noticed it is in a separate folder) and there are many things not implemented. So yes, this heap memory has to be deleted and will be done in `Connection` class which will be the one holding the reference to these instances, you'll see it in the next PRs!